doc/src/advanced/potential.md

```diff
@@ -0,0 +1,254 @@
+
+Let's start by having a look at the API documentation for `Potential`. Open the [API documentation](http://lumol.org/lumol/latest/lumol/index.html) and navigate to the `energy` module which is listed under the `modules` section at the bottom of the page. Traits are listed at the very bottom of the module documentation. Open the documentation for the `Potential` trait.
+
+As you can see from the trait, a `Potential` defines the interface for two functions, `energy` and `force` (we ignore the `Sync + Send` statement for now):
+
+```rust
+pub trait Potential: Sync + Send {
+    fn energy(&self, x: f64) -> f64;
+    fn force(&self, x: f64) -> f64;
+}
+```
+
+It's pretty self-explanatory what these to functions do: `energy` will compute the interaction energy between two sites as a function of `x`. The force is defined as the negative derivative of the energy function with respect to `x`.
+
+Both functions take a single, scalar argument and return a single scalar value. In our case `x` stands for the distance between two interaction sites. Note that only the function definitions -- without a function body -- are specified. We will have to implement these functions for our potential.
+
+Let's start the implementation.
+
+## Adding functionality and documentation into the code
+
+### Defining the struct
+
+The energy function of the Mie potential reads
+
+$u(x) = \frac{n}{n-m} \left(\frac{n}{m}\right)^{m/(n-m)}\epsilon \left[ \left( \frac{\sigma}{x}\right)^n - \left( \frac{\sigma}{x}\right)^m \right]$
+
+where $x$ denotes the distance between two interaction sites $i, j$, $x = x_{ij} = | \mathbf{r}_j - \mathbf{r}_i |$. The parameters of the potential are
+
+- $n, m$ the repulsive and attractive exponents, respectively,
+- $\epsilon$ the energetic paramater,
+- $\sigma$ the particle diameter or structural parameter.
+
+We start by defining the `struct` for our potential. We will add our code to the `functions.rs` file located in `lumol/src/core/src/energy`. Add the following lines:
+
+```rust
+#[derive(Clone, Copy)]
+pub struct Mie {
+    /// Distance constant
+    pub sigma: f64,
+    /// Energy constant
+    pub epsilon: f64,
+    /// Exponent of repulsive contribution
+    pub n: f64,
+    /// Exponent of attractive contribution
+    pub m: f64,
+    /// Energetic prefactor computed from the exponents and epsilon
+    pub prefac: f64,
+}
+```
+
+Next, we implement a constructor function. This is usefull in this case since we want to compute the prefactor
+of the potential once before we start our simulation.
+In Rust we typically use `new` for the constructors' name.
+
+```rust
+impl Mie {
+    fn new(sigma: f64, epsilon: f64, n: f64, m: f64) -> Mie {
```

Comment:
If the `Mie::new` function is enforcing some constraint, maybe the fields should not be `pub`lic